src/libraries/Common/src/System/Security/Cryptography/KeyFormatHelper.Encrypted.cs

```diff
@@ -272,6 +274,68 @@ internal static ArraySegment<byte> DecryptPkcs8(
                 out bytesRead);
         }
 
+        internal static unsafe T DecryptPkcs8<T>(
+            ReadOnlySpan<char> password,
+            ReadOnlySpan<byte> source,
+            ReadOnlySpanFunc<byte, T> keyReader,
+            out int bytesRead)
+        {
+            fixed (byte* pointer = source)
+            {
+                using (PointerMemoryManager<byte> manager = new(pointer, source.Length))
+                {
+                    ArraySegment<byte> decrypted = KeyFormatHelper.DecryptPkcs8(password, manager.Memory, out bytesRead);
+                    AsnValueReader reader = new(decrypted, AsnEncodingRules.BER);
+                    reader.ReadEncodedValue();
```

Comment:
These lines can't throw, or we don't want to return decrypted to the pool if they do?

Reply:
Fixed.  I think I based this on some other code where it had run through Decode to know it was structurally valid, and I just needed to know if it had trailing data.  But since this hasn't, those should be in the try.